include/fmt/format.h

```diff
@@ -742,9 +742,14 @@ void basic_memory_buffer<T, SIZE, Allocator>::grow(size_t size) {
 #ifdef FMT_FUZZ
   if (size > 5000) throw std::runtime_error("fuzz mode - won't grow that much");
 #endif
+  const size_t max_size = std::allocator_traits<Allocator>::max_size(alloc_);
+  if (size > max_size) {
+    throw std::length_error("fmt::basic_memory_buffer::grow");
+  }
```

Comment:
Please remove this to reduce the number of checks.

Reply:
Removed.